packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4734,11 +4734,60 @@ class Listener extends SingleChildRenderObjectWidget {
 ///
 /// This widget creates a separate display list for its child, which
 /// can improve performance if the subtree repaints at different times than
-/// the surrounding parts of the tree. Specifically, when the child does not
-/// repaint but its parent does, we can re-use the display list we recorded
-/// previously. Similarly, when the child repaints but the surround tree does
-/// not, we can re-record its display list without re-recording the display list
-/// for the surround tree.
+/// the surrounding parts of the tree.
+///
+/// This is useful since [RenderObject.paint] may be triggered even if its
+/// associated [Widget] instances' did not change or rebuild. A [RenderObject]
+/// will repaint whenever any [RenderObject] that shares the same [Layer] is
+/// marked as being dirty and needing paint (see [RenderObject.markNeedsPaint]).
+///
+/// such as when an
+/// ancestor scrolls or when an ancestor or descendent's widget is going through
+/// an animation.
+///
+/// Containing [RenderObject.paint] to parts of the render subtree that are
+/// actually visually changing using [RepaintBoundary] explicitly or implicitly
+/// is therefore critical to minimizing redundant work and improving the app's
+/// performance. Render tree repaints can be monitored via debug flags such as
+/// [debugRepaintRainbowEnabled] or [debugProfilePaintsEnabled].
+///
+/// When a [RenderObject] is flagged as needing to paint via
+/// [RenderObject.markNeedsPaint], the nearest ancestor [RenderObject] up to
+/// possibly the root of the application with [RenderObject.isRepaintBoundary]
+/// is alerted to repaint. That nearest ancestor's repaint process will cause
```

Comment:
is requested to repaint. "alerted to" something means "told about"; "told about repaint" doesn't really make sense.

Reply:
done